contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,631 @@
+# Local Storage Persistent Volumes
+
+Authors: @msau42, @vishh
+
+This document presents a detailed design for supporting persistent local storage,
+as outlined in [Local Storage Overview](local-storage-overview.md).
+Supporting all the use cases for persistent local storage will take many releases,
+so this document will be extended for each new release as we add more features.
+
+## Goals
+
+* Allow pods to mount any local block or filesystem based volume.
+* Allow pods to mount dedicated local disks, or channeled partitions as volumes for
+IOPS isolation.
+* Allow pods do access local volumes without root privileges.
+* Allow pods to access local volumes without needing to undestand the storage
+layout on every node.
+* Persist local volumes and provide data gravity for pods.  Any pod
+using the local volume will be scheduled to the same node that the local volume
+is on.
+* Allow tolerations on volumes to be specified to release the volume during
+failure conditions, such as node, storage or scheduling failures.
+* Allow pods to specify local storage as part of a Deployment or StatefulSet.
+* Allow Administrators to set up and configure local volumes with simple methods.
+* Do not require qdministrators to manage the local volumes once provisioned
+for a node.
+
+## Non-Goals
+
+* Provide data availability for a local volume beyond its local node.
+* Support the use of HostPath volumes and Local PVs on the same volume.
+
+## Background
+
+In Kubernetes, there are two main types of storage: local and remote.
+Local storage is typically used with ephemeral volumes where the data only
+persists during the lifetime of the pod.  Remote storage is typically used with
+persistent volumes where the data can persist beyond the lifetime of the pod.
+
+There is increasing demand for using local storage with persistent volumes,
+especially for distributed filesystems and databases such as GlusterFS and
+Cassandra.  The main motivations for using local storage include:
+
+* Performance:  Local SSDs achieve higher IOPS and throughput than a cloud
+provider’s remote storage.
+
+* Cost: High-performance networked storage is expensive to setup and maintain.
```

Comment:
if the upper layer application already does replication (and is not file/block level replication aware), it usually makes sense to just use local disk and let the application take over replication instead of using replicated block device.

Reply:
Good point.  I think I mentioned it already in the distributed db and fs use case, but will add it here too.